src/ext/oblivc/obliv_bits.c

```diff
@@ -364,25 +384,53 @@ void protocolAddSizeCheck(ProtocolDesc* pd)
   t->cb.recv=sizeCheckRecv;
   t->cb.cleanup=sizeCheckCleanup;
 }
-// ---------------------------------------------------------------------------
 
-void cleanupProtocol(ProtocolDesc* pd)
-  { pd->trans->cleanup(pd->trans); }
+// --------------------------- Protocols -----------------------------------
 
+int ocCurrentParty() { return currentProto->currentParty(currentProto); }
+int ocCurrentPartyDefault(ProtocolDesc* pd) { return pd->thisParty; }
 void setCurrentParty(ProtocolDesc* pd, int party)
   { pd->thisParty=party; }
 
-void __obliv_c__assignBitKnown(OblivBit* dest, bool value)
-  { dest->knownValue = value; dest->unknown=false; }
+ProtocolDesc* ocCurrentProto() { return currentProto; }
+void ocSetCurrentProto(ProtocolDesc* pd) { currentProto=pd; }
 
-void __obliv_c__copyBit(OblivBit* dest,const OblivBit* src)
-  { if(dest!=src) *dest=*src; }
+void ocSplitProto(ProtocolDesc* pdout, ProtocolDesc * pdin)
+{
+  *pdout = (struct ProtocolDesc) {
+    .error = pdin->error,
+    .partyCount = pdin->error,
+    .currentParty = pdin->currentParty,
+    .feedOblivInputs = pdin->feedOblivInputs,
+    .revealOblivBits = pdin->revealOblivBits,
+    .setBitAnd = pdin->setBitAnd,
+    .setBitOr = pdin->setBitOr,
+    .setBitXor = pdin->setBitXor,
+    .setBitNot = pdin->setBitNot,
+    .flipBit = pdin->flipBit,
+    .thisParty = pdin->thisParty,
+    .trans = pdin->trans->split(pdin->trans),
+    .splitextra = pdin->splitextra,
+    .cleanextra = pdin->cleanextra,
+    .extra = NULL
+  };
+  if (pdout->splitextra != NULL && pdin->extra != NULL) pdout->splitextra(pdout, pdin);
+  oflush(pdin); oflush(pdout);
```

Comment:
Do we need to flush both pdin and pdout? More importantly, isn't this step Yao-specific, and should go into splitextra()?